Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift

```diff
@@ -12,8 +12,11 @@ import StitchCoreSDK
  */
 internal func docConfigFilter(forNamespace namespace: MongoNamespace,
                               withDocumentId documentId: AnyBSONValue) -> Document {
-    return [CoreDocumentSynchronization.CodingKeys.namespace.rawValue:
-        try? BSONEncoder().encode(namespace),
+    guard let namespaceDoc = try? BSONEncoder().encode(namespace) else {
+        return [CoreDocumentSynchronization.CodingKeys.namespace.rawValue: BSONNull(),
+         CoreDocumentSynchronization.CodingKeys.documentId.rawValue: documentId.value]
+    }
+    return [CoreDocumentSynchronization.CodingKeys.namespace.rawValue: namespaceDoc,
```

Comment:
[style nit] could we get rid of the guard block and simplify this to `try? BSONEncoder().encode(namespace) ?? BSONNull()`?

Reply:
No, we cannot. This applies to all of your comments. The compiler doesn't like it.